broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java

```diff
@@ -152,6 +162,24 @@ public int getValidSnapshotsCount() {
     return constructableSnapshotStore.getLatestSnapshot().isPresent() ? 1 : 0;
   }
 
+  private void tryCompactState() throws Exception {
+    final var filesBeforeCompaction = runtimeDirectory.toFile().listFiles().length;
+
+    if (filesBeforeCompaction < triggerFileCountManualCompaction) {
+      return;
+    }
+    LOG.debug(
+        "Running forced compaction of the state in {} with {} files.",
+        runtimeDirectory,
+        filesBeforeCompaction);
+    db.compactFiles(outputLevelManualCompaction);
+    final var filesAfterCompaction = runtimeDirectory.toFile().listFiles().length;
+    LOG.debug(
```

Comment:
In this case (and above), could we argue this should be info level? Wouldn't a normal user want to know about this?

Reply:
:+1: